check_binary.sh

```diff
@@ -0,0 +1,194 @@
+# Python returned by `which python`). Or, if this is testing a libtorch
+# Pythonless binary, then it expects to be in the root folder of the unzipped
+# libtorch package. 
+
+
+# The install root depends on both the package type and the os
+# All MacOS packages use conda, even for the wheel packages.
+if [[ "$PACKAGE_TYPE" == libtorch ]]; then
+  install_root="$pwd"
+else
+  py_dot="${DESIRED_PYTHON:0:3}"
+  install_root="$(dirname $(which python))/../lib/python${py_dot}/site-packages/torch/"
+fi
+
+
+###############################################################################
+# Check GCC ABI
+###############################################################################
+echo "Checking that the gcc ABI is what we expect"
+if [[ "$(uname)" != 'Darwin' ]]; then
+  function is_expected() {
+    if [[ "$DESIRED_DEVTOOLSET" == 'devtoolset7' ]]; then
+      if [[ "$1" == 1 || "$1" == "ON" ]]; then
+        echo 1
+      fi
+    else
+      if [[ "$1" == 0 || "$1" == "OFF" ]]; then
+        # TODO is it also okay if this is empty?
+        echo 1
+      fi
+    fi
+  }
+
+  # First we check that the env var in TorchConfig.cmake is correct
+
+  # We search for D_GLIBCXX_USE_CXX11_ABI=1 in torch/TorchConfig.cmake
+  torch_config="${install_root}/share/cmake/Torch/TorchConfig.cmake"
+  if [[ ! -f "$torch_config" ]]; then
+    echo "No TorchConfig.cmake found!"
+    ls -lah "$install_root/share/cmake/Torch"
+    exit 1
+  fi
+  echo "Checking the TorchConfig.cmake"
+  cat "$torch_config"
+
+  # The sed call below is
+  #   don't print lines by default (only print the line we want)
+  # -n
+  #   execute the following expression
+  # e
+  #   replace lines that match with the first capture group and print
+  # s/.*D_GLIBCXX_USE_CXX11_ABI=\(.\)".*/\1/p
+  #   any characters, D_GLIBCXX_USE_CXX11_ABI=, exactly one any character, a
+  #   quote, any characters
+  #   Note the exactly one single character after the '='. In the case that the
+  #     variable is not set the '=' will be followed by a '"' immediately and the
+  #     line will fail the match and nothing will be printed; this is what we
+  #     want.  Otherwise it will capture the 0 or 1 after the '='.
+  # /.*D_GLIBCXX_USE_CXX11_ABI=\(.\)".*/
+  #   replace the matched line with the capture group and print
+  # /\1/p
+  actual_gcc_abi="$(sed -ne 's/.*D_GLIBCXX_USE_CXX11_ABI=\(.\)".*/\1/p' < "$torch_config")"
+  if [[ "$(is_expected "$actual_gcc_abi")" != 1 ]]; then
+    echo "gcc ABI $actual_gcc_abi not as expected."
+    exit 1
+  fi
+
+  # We also check that there are[not] cxx11 symbols in libtorch
+  echo "Checking that symbols in libcaffe2.so have the right gcc abi"
+  libtorch="${install_root}/lib/libcaffe2.so"
+  set +e
+  cxx11_symbols="$(nm "$libtorch" | c++filt | grep cxx11)"
+  has_cxx11_symbols="$?"
+  set -e
+  if [[ "$(is_expected "$has_cxx11_symbols")" != 1 ]]; then
+    if [[ "$has_cxx11_symbols" == 0 ]]; then
```

Comment:
I think this is backwards. Doesn't `$?` give the return code, which would be 0 when symbols are found

Reply:
Yes :(